src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/FSharp/FSharpProjectSelector.cs

```diff
@@ -38,5 +41,22 @@ internal static void GetProjectFactoryGuid(XDocument doc, out Guid guidProjectFa
 
             guidProjectFactory = ProjectType.LegacyFSharpGuid;
         }
+
+        public static async Task<IDisposable> RegisterAsync(ManagedProjectSystemPackage package)
+        {
+            IVsRegisterProjectSelector projectSelector = await package.GetServiceAsync<SVsRegisterProjectTypes, IVsRegisterProjectSelector>();
+            
```

Comment:
This needs an explicit switch to the UI thread.

Reply:
In the latest update I've documented on the interface that all calls to `IPackageService.InitializeAsync` occur on the UI thread.